src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -0,0 +1,571 @@
+    ):
+        env, vol = self.setup_callback_server_envs(job_dir, kb_base_url, token, port)
+        self.container = client.containers.run(
+            name=container_name,
+            image=loader_common_names.CALLBACK_UPLOADER_IMAGE_NAME,
+            detach=True,
+            network_mode="host",
+            environment=env,
+            volumes=vol,
+        )
+        time.sleep(2)
+
+    def stop_callback_server(self):
+        self.container.stop()
+        self.container.remove()
+
+
+def _get_parser():
+    parser = argparse.ArgumentParser(
+        description="PROTOTYPE - Upload assembly files to the workspace service (WSS).",
+        formatter_class=loader_helper.ExplicitDefaultsHelpFormatter,
+    )
+
+    required = parser.add_argument_group("required named arguments")
+    optional = parser.add_argument_group("optional arguments")
+
+    # Required flag argument
+    required.add_argument(
+        "--workspace_id",
+        required=True,
+        type=int,
+        help="Target workspace addressed by the permanent ID",
+    )
+    required.add_argument(
+        f"--{loader_common_names.KBASE_COLLECTION_ARG_NAME}",
+        type=str,
+        help="The name of the collection being processed",
+    )
+    required.add_argument(
+        f"--{loader_common_names.SOURCE_VER_ARG_NAME}",
+        type=str,
+        help="The source version of the collection being processed",
+    )
+
+    # Optional argument
+    optional.add_argument(
+        "--root_dir",
+        type=str,
+        default=loader_common_names.ROOT_DIR,
+        help="Root directory.",
+    )
+    optional.add_argument(
+        "--token_filepath",
+        type=str,
+        help="A file path that stores a KBase token",
+    )
+    optional.add_argument(
+        "--env",
+        type=str,
+        choices=loader_common_names.KB_ENV,
+        default="PROD",
+        help="KBase environment, defaulting to PROD",
+    )
+    optional.add_argument(
+        "--upload_file_ext",
+        type=str,
+        nargs="+",
+        default=UPLOAD_FILE_EXT,
+        help="Upload only files that match given extensions",
+    )
+    optional.add_argument(
+        "--workers",
+        type=int,
+        default=5,
+        help="Number of workers for multiprocessing",
+    )
+    optional.add_argument(
+        "--keep_job_dir",
+        action="store_true",
+        help="Keep SDK job directory after upload task is completed",
+    )
+    return parser
+
+
+def _get_yaml_file_path(assembly_dir: str) -> str:
+    """
+    Get the uploaded.yaml file path from collections source directory.
+    """
+    file_path = os.path.join(assembly_dir, UPLOADED_YAML)
+    Path(file_path).touch(exist_ok=True)
+    return file_path
+
+
+def _sanitize_data_dir(job_dir: str) -> str:
+    """
+    Create a temporary directory for storing uploaded files.
+    """
+    data_dir = os.path.join(job_dir, "workdir/tmp", DATA_DIR)
```

Comment:
Why workdir/tmp?

Reply:
So looking at this yet again, I don't understand how this is working. We're mounting the external job dir (e.g. `/global/cfs...` etc.) to the same dir path inside the container:

```
vol[job_dir] = {"bind": job_dir, "mode": "rw"}
```

... but `AssemblyUtil` writes to `/kb/module/workdir/tmp`, so the files shouldn't be visible in the mount because the write dir and the mounted dir are different.

Am I missing something here?

Reply:
Oh, there's a `JOB_DIR` env var that is also set to `job_dir`. I'm guessing the job runner / callback server sets the scratch directory to `<JOB_DIR>/workdir/tmp`. So `JOB_DIR` is shared between the host and container but the actual SDK app sees `<JOB_DIR>/workdir/tmp` as its work area

Reply:
In that case I'd make a single function shared between the uploader and downloader for creating the `<JOB_DIR>/workdir/tmp` directory so we don't have the same constant value in two places

Reply:
maybe that could be done in the Conf constructor

Reply:
Actually, no, I'm wrong again, because JOB_DIR_IN_ASSYUTIL_CONTAINER (or whatever) is `/kb/module/workdir/tmp`. So I'm back to not understanding what's going on here

Reply:
Ugh, that volume mount is for the callback server, not the AssemblyUtils container. So the JOB_DIR variable to the CBS tells it where on the host to mount `/kb/module/workdir/tmp`. 

I don't understand the purpose of that mount for the CBS though.

Reply:
> I don't understand the purpose of that mount for the CBS though.

Oh, probably so the CBS can write out the token / config / etc.

Reply:
I made a single function shared between the uploader and downloader and initiated inside the Conf constructor now.

I made it work mainly through trial and error when I first tested `save_assembly_from_fasta`. I kept getting target file is not found error, and that's when I logged in the AssemblyUtils container and tweaked stuff.

What I saw is the AssemblyUtils container(`/kb/module/work/tmp`) shares volume with the callback server container(`<JOB_DIR>/workdir/tmp`). We need to confirm with Shane what we see is same as how it should work.

Reply:
> What I saw is the AssemblyUtils container(/kb/module/work/tmp) shares volume with the callback server container(<JOB_DIR>/workdir/tmp). We need to confirm with Shane what we see is same as how it should work.

That's what I would expect given how the SDK works and the code in `Conf`

Reply:
:+1: